packages/schematics/angular/utility/ast-utils.ts

```diff
@@ -93,9 +93,11 @@ export function insertImport(source: ts.SourceFile, fileToEdit: string, symbolNa
  * @param node
  * @param kind
  * @param max The maximum number of items to return.
+ * @param recursive Continue looking for nodes of kind recursive until end
+ * the last child even when node of kind has been found.
  * @return all nodes of kind, or [] if none is found
  */
-export function findNodes(node: ts.Node, kind: ts.SyntaxKind, max = Infinity): ts.Node[] {
+export function findNodes(node: ts.Node, kind: ts.SyntaxKind, max = Infinity, recursive = false): ts.Node[] {
```

Comment:
Should the default be the opposite so that existing calls retain the same behavior?  Or was that problematic for the other callers as well?

Reply:
Yeah, this is problematic in other calls like `insertAfterLastOccurrence`. Since when have nested nodes of the same kind. Example: `[[]], {a: {}}` the nodes returned will not be the correct once.

I don't thing we ever would want to get the recursive nested nodes of a kind (I might be wrong). When for example we are looking for `ObjectLiteralExpression` we would typically want the "direct" `ObjectLiteralExpression` and the the direct `ObjectLiteralExpression` and the nested ObjectLiteralExpression .

Reply:
I was actually tempted to remove the recursion for good.

Reply:
Makes sense.  Just wanted to make sure.